lms/services/blackboard_api/service.py

```diff
@@ -53,12 +53,38 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        return [
+            {
+                "id": self._blackboard_document_url(file_["id"]),
+                "display_name": file_["name"],
+                "updated_at": file_["modified"],
+            }
+            for file_ in files
+        ]
+
+    def public_url(self, course_id, file_id):
+        files = self._http_service.get(
+            self._api_url(f"/courses/uuid:{course_id}/resources"),
+            oauth=True,
+            schema=BlackboardListFilesSchema,
+        ).validated_data
+
+        for file_ in files:
+            if self._blackboard_document_url(file_["id"]) == file_id:
+                return file_["downloadUrl"]
```

Comment:
To get a public URL we actually have to call the _list files_ API and then iterate over the list of file dicts looking for the one with the ID that we're looking for. The download URLs are in the file dicts. This sucks because on every student launch we might have to paginate through multiple responses to get the full list of files. But I'm not sure if the Blackboard API gives us any other option. I need to take a look at their API docs and see if there's any other way.

Note that we format the IDs from the API into `blackboard://...` URLs. Rather than parsing the `file_id` argument (which is already in `blackboard://...` URL format) into a plain ID. This means that we only ever need for format `blackboard://...` URLs, we never parse them. (Yes, we are wasting some CPU cycles :man_shrugging: )

Reply:
Mentioned elsewhere but the `file_id` argument should maybe be renamed `document_url` since it's a `blackboard://...` URL